Extensions/UITextField.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  UITextField.swift
+//  LoopKitUI
+//
+//  Created by Rick Pasetto on 9/17/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import UIKit
+
+extension UITextField {
+    
+    func selectAll(completion: (() -> Void)? = nil) {
+        DispatchQueue.main.async {
```

Comment:
Are these dispatches just to ensure things get run on the main queue, or do you need to defer processing generally for some reason?  If the the former, they should be replaced with assertions, or just removed; as there is a strong assumption already in place for methods on UI components that they are used from the main thread.

Reply:
If clients are in a position where they might be calling these methods from a different queue, I think it is their responsibility to dispatch to main. Generally I would expect clients to already be on .main, so this would be unneeded overhead.

Reply:
It's the former, or else I see runtime warnings.

I'm fine with switching these to assertions and moving the dispatching to the callers.  I'll do that 👍

Reply:
Oooh... I take that back (and I forgot about this...I confused it with some other work I was doing).
Without hopping off onto `.main` to change the selection, the selection doesn't change.  (I honestly don't know why).

Since you approved, @ps2  , I presume you are ok with it as-is?

Reply:
Yeah, I was wondering if something like this was happening. This also feels like a client responsibility.  The issue is probably that you can't set selectedTextRange from within a textFieldDidBeginEditing callback, but would be fine to set from other code. So in this case I would still recommend dispatching in the client (if the client happens to be in this kind of callback).

Reply:
Ah ok, a fair point.  I know I already merged this, but I'll work on changing it.  Thanks.